### PR TITLE
Limit PSP removal patch to vintage providers

### PR DESCRIPTION
**Pull request by kubasobon**
Towards giantswarm/roadmap#2716

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

**Comment by kubasobon** [comment]
Mr @stone-z, it's ready.

**Comment by kubasobon** [comment]
@AverageMarcus this will hopefully prevent CAPx clusters from experiencing issues.

**Review by mproffitt — APPROVED** [review]

File: main.go [hunk 1]
```diff
@@ -63,6 +63,7 @@ func mainWithError() error {
 		c := app.MutatorConfig{
 			K8sClient: cfg.K8sClient,
 			Logger:    newLogger,
+			Provider:  cfg.Provider,
```
Comment (stone-z) [inline main.go:66]: Where does the provider actually come from?

**Review by stone-z — APPROVED** [review]
This LGTM. Convenient the provider is already there :+1:

File: main.go (hunk 1 above)
Comment (stone-z) [reply main.go:66]: Nvm, found it

Comment (kubasobon) [reply main.go:66]: It's a required command-line parameter. Defined in the [config repo](https://github.com/giantswarm/config/blob/main/default/apps/app-admission-controller/configmap-values.yaml.template#L4-L5), passed on in [Deployment's `args`](https://github.com/giantswarm/app-admission-controller/blob/master/helm/app-admission-controller/templates/deployment.yaml#L37-L41).

**Pull request merged**